addons/stock/models/stock_move.py

```diff
@@ -2208,3 +2208,44 @@ def _convert_string_into_field_data(self, string, options):
         if regex_findall(r'^([0-9]+\.?[0-9]*|\.[0-9]+)$', string):  # Number => Quantity.
             return {'quantity': float(string)}
         return False
+
+    def _match_searched_availability(self, operator, value, _get_comparison_date):
+        def get_stock_moves(moves, state):
+            if state == 'available':
+                return moves.filtered(lambda m: m.forecast_availability == m.product_qty and not m.forecast_expected_date)
+            elif state == 'expected':
+                return moves.filtered(lambda m: m.forecast_availability == m.product_qty and m.forecast_expected_date and m.forecast_expected_date <= _get_comparison_date(m))
+            elif state == 'late':
+                return moves.filtered(lambda m: m.forecast_availability == m.product_qty and m.forecast_expected_date and m.forecast_expected_date > _get_comparison_date(m))
```

Comment:
I think the underscore before a names means either the function is private, either the variable is not used. It's not the case here and not needed here
```suggestion
    def _match_searched_availability(self, operator, value, get_comparison_date):
        def get_stock_moves(moves, state):
            if state == 'available':
                return moves.filtered(lambda m: m.forecast_availability == m.product_qty and not m.forecast_expected_date)
            elif state == 'expected':
                return moves.filtered(lambda m: m.forecast_availability == m.product_qty and m.forecast_expected_date and m.forecast_expected_date <= get_comparison_date(m))
            elif state == 'late':
                return moves.filtered(lambda m: m.forecast_availability == m.product_qty and m.forecast_expected_date and m.forecast_expected_date > get_comparison_date(m))
```